docs/coding_style.md

```diff
@@ -0,0 +1,190 @@
+# Coding Conventions
+
+
+## Background
+
+To improve the quality and maintainability of the code, we summarized some common coding standards and conventions.
+
+There are many style guides, and they may conflict with each other. To avoid overly arguing formatting, we make decisions based on the following priorities:
+
+- [Google Python Style](https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings), [PEP 8](https://peps.python.org/pep-0008/)
+- Framework Style
+- Internal Style
+- Sub-module specific Style
+
+## Rules
+
+> Note: The sub-tile naming is following [Google Python Style](https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings) and [PEP 8](https://peps.python.org/pep-0008/). See the relevant section for more details.
+
+
+### Imports
+
+- Recommend
+
+```python
+import os
+import sys
+
+from x import py
+from x import y as z
+from copy import deepcopy
+from subprocess import Popen, PIPE
+```
+
+- Not recommend
+
+```python
+from sub_module import *  # May lead to namespace pollution
+import os, sys  # Import on separate lines
+import copy  # May import local copy.py
```

Comment:
copy is a built in module https://docs.python.org/3/library/copy.html ?

Reply:
This means we recommend to only import the module used by the code rather than import the whole package.

Reply:
I don't think copy has any submodules iirc? There is a risk of this line being misleading because there's no way to import the copy module otherwise. Please correct me if I am missing anything